cli/src/cmd/utils.rs

```diff
@@ -299,3 +307,114 @@ pub fn read_constructor_args_file(constructor_args_path: PathBuf) -> eyre::Resul
     };
     Ok(args)
 }
+
+/// A slimmed down return from the executor
+pub struct TraceResult {
+    pub success: bool,
+    pub traces: Traces,
+    pub debug: DebugArena,
+    pub gas_used: u64,
+}
+
+/// labels the traces, conditonally prints them or opens the debugger
+pub async fn handle_traces(
+    mut result: TraceResult,
+    config: &Config,
+    chain: Option<ethers::types::Chain>,
+    labels: Vec<String>,
+    verbose: bool,
+    debug: bool,
+) -> eyre::Result<()> {
+    let mut etherscan_identifier = EtherscanIdentifier::new(config, chain)?;
+
+    let labeled_addresses = labels.iter().filter_map(|label_str| {
+        let mut iter = label_str.split(':');
+
+        if let Some(addr) = iter.next() {
+            if let (Ok(address), Some(label)) =
+                (ethers::types::Address::from_str(addr), iter.next())
+            {
+                return Some((address, label.to_string()));
+            }
+        }
+        None
+    });
+
+    let mut decoder = CallTraceDecoderBuilder::new().with_labels(labeled_addresses).build();
+
+    decoder.add_signature_identifier(SignaturesIdentifier::new(
+        Config::foundry_cache_dir(),
+        config.offline,
+    )?);
+
+    for (_, trace) in &mut result.traces {
+        decoder.identify(trace, &mut etherscan_identifier);
+    }
+
+    if debug {
+        let (sources, bytecode) = etherscan_identifier.get_compiled_contracts().await?;
+        run_debugger(result, decoder, bytecode, sources)?;
+    } else {
+        print_traces(&mut result, decoder, verbose).await?;
+    }
+
+    Ok(())
+}
+
+pub async fn print_traces(
+    result: &mut TraceResult,
+    decoder: CallTraceDecoder,
+    verbose: bool,
+) -> eyre::Result<()> {
+    if result.traces.is_empty() {
+        eyre::bail!("Unexpected error: No traces. Please report this as a bug: https://github.com/foundry-rs/foundry/issues/new?assignees=&labels=T-bug&template=BUG-FORM.yml");
```

Comment:
It's aight to error like this, but I think we can also just panic with `expect` and the hook will catch it and add the "this is a bug. please report" part :D

Reply:
i changed it to a panic!